timings/state.go

```diff
@@ -121,3 +129,37 @@ func (t *Timings) Save(st *state.State) {
 	}
 	st.Set("timings", stateTimings)
 }
+
+// Get returns timings for which filter predicate is true and filters out nested timings whose level is greater than maxLevel.
+// Negative maxLevel value disables filtering by level.
+// It's responsibility of the caller to lock the state before calling this function.
+func Get(st *state.State, maxLevel int, filter func(tags map[string]string) bool) ([]*TimingsInfo, error) {
+	var stateTimings []rootTimingsJSON
+	if err := st.Get("timings", &stateTimings); err != nil && err != state.ErrNoState {
+		return nil, fmt.Errorf("could not get timings data from the state: %v", err)
+	}
+
+	var result []*TimingsInfo
+	for _, tm := range stateTimings {
+		if !filter(tm.Tags) {
+			continue
+		}
+		res := &TimingsInfo{
+			Tags: tm.Tags,
+		}
+		if maxLevel < 0 { // no level-based filtering?
+			res.NestedTimings = tm.NestedTimings
+			result = append(result, res)
+			continue
+		}
+		for _, nested := range tm.NestedTimings {
+			if nested.Level <= maxLevel {
+				res.NestedTimings = append(res.NestedTimings, nested)
+			}
+		}
+		if len(res.NestedTimings) > 0 {
```

Comment:
is this if tested?

Reply:
This if is not needed after all, it's always true. I've added comment in the code explaining that.